### PR TITLE
Add default scatter marker option to rcParams

### PR DESCRIPTION
**Pull request by benjamincongdon**
Addresses #6272.

Removes hard-coded `'o'` as the default marker from within the `scatter` definition. Adds the `'scatter.marker'` option in `rcParams`, which is then loaded when `scatter` is called.

This is my first contribution, so I apologize in advance if I've violated any best practices.

### CODE REVIEW

**Comment by benjamincongdon** [comment]
The AppVeyor build looks like it failed because it took too long to download freetype.

**Comment by tacaswell** [comment]
Can you also add this key to `matplotlibrc.template` and the classic style sheet?

**Comment by benjamincongdon** [comment]
Alright, done.

File: matplotlibrc.template [hunk 1]
```diff
@@ -461,6 +461,9 @@ backend      : $TEMPLATE_BACKEND
                                   # If Numpy 1.11 or later is
                                   # installed, may also be `auto`
 
+### SCATTER PLOTS
+#scatter.marker                   # The default marker type for scatter plots.
```
Comment (tacaswell) [inline matplotlibrc.template]: This needs `: o`

**Comment by benjamincongdon** [comment]
Whoops, my mistake. Fixed.

**Comment by tacaswell** [comment]
One more house keeping thing, can you add a note in https://github.com/matplotlib/matplotlib/blob/master/doc/users/whats_new/rcparams.rst documenting this feature?

**Comment by benjamincongdon** [comment]
Sure, added that to the docs.

**Comment by tacaswell** [comment]
Thanks!  backported to v2.x as ad81e3f

I think this is your first contribution to mpl.  Congratulations, hopefully we will see you again :smile:

**Pull request merged**